### PR TITLE
Feature/transaction nonce

### PR DESCRIPTION
**Pull request by dabasov**
A brief description of the changes in this PR:

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/zboxcli/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

Associated PRs (Link as appropriate):
- 0chain: 
- blobber:
- gosdk:
- system_test:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -49,6 +50,7 @@ func init() {
 	rootCmd.PersistentFlags().StringVar(&walletFile, "wallet", "", "wallet file (default is wallet.json)")
 	rootCmd.PersistentFlags().StringVar(&walletClientID, "wallet_client_id", "", "wallet client_id")
 	rootCmd.PersistentFlags().StringVar(&walletClientKey, "wallet_client_key", "", "wallet client_key")
+	rootCmd.PersistentFlags().Int64Var(&nonce, "withNonce", 0, "nonce that will be used in transaction (default is 0)")
```
Comment (peterlimg) [inline cmd/root.go:53]: Isn't nonce mandatory? I think user should not be allowed to run cli without nonce

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Comment by peterlimg** [comment]
@dabasov there too many commits for merging code from staging, the commit history will be clearer if we rebase and merge those commits from staging.

File: cmd/root.go (hunk 1 above)
Comment (dabasov) [reply cmd/root.go:53]: nonce can be optionally set with zboxcli, if not it will be queried by gosdk

**Comment by dabasov** [comment]
> 

Can we do it right in the github merging pull request? there is squash commits feature, think it can work in this case

**Comment by peterlimg** [comment]
yes, sure, we can do it in github

File: cmd/root.go (hunk 1 above)
Comment (peterlimg) [reply cmd/root.go:53]: okay

**Review by peterlimg — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | feature/transaction-nonce  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | feature/transaction-nonce  |
| zwalletcli  | feature/transaction-nonce  |
| 0chain  | feature/transaction_nonce  |
| blobber  | feature/transaction-nonce  |
| authorizer  | staging  |
| 0box  | feature/transaction-nonce  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**